cbpi/configFolder.py

```diff
@@ -115,19 +165,24 @@ def create_config_file(self):
         print("Config Folder created")
 
     def create_home_folder_structure(configFolder):
-        pathlib.Path(os.path.join(".", 'logs/sensors')).mkdir(parents=True, exist_ok=True)
```

Comment:
i just noticed that this was intentional. the logs are of course not meant to be inside the config folder but in the folder wherever the user did `cbpi setup`. Im going to change this back.

Reply:
should we really put the logs where `cbpi setup` is called? Then we have to call `cbpi start` from the same folder every time or the log folder is in the wrong place.

i would rather place the log folder in the parent folder of the provided config. This makes more sense especially when we can provide a config folder path somewhere completely different.

For example in the devcontainer setup this would mean the logs folder is inside the .devcontainer folder instead of the repository root.

I will implement the log folder to always be in the parent directory of the provided config folder if there are no objections.

Reply:
How about a separate config variable for the log location similar to the config directory now?

Reply:
i was thinking about this as well but it would break portability of the config. If you provide me with a zipped config for me to test/debug something, I restore it in my docker and then the check setup fails because i dont have the path /user/home/papauorg/cbpi/logs in my docker container. So i dont even get to logging and need to manually alter config files.

its the same when i transfer my own real setup to my docker container to test soemthing.

Reply:
> i was thinking about this as well but it would break portability of the config

I don't think it would break portability of the config because the logs are not part of the config (or at least should not be). 

> If you provide me with a zipped config for me to test/debug something, I restore it in my docker and then the check setup fails because i dont have the path /user/home/papauorg/cbpi/logs in my docker container.

I didn't mean that the path to the log directory should be written anywhere in the config-files. What I meant was, that you can start cbpi with a parameter that specifies where the logs should be written to. I don't really understand why the log directory is relevant in your sample. All you need to run cbpi would be the config. And I'd avoid absolute paths whenever possible.

If you can not start cbpi if the log directory is not present, maybe that's something that should be changed. Either create the log directory if it does not exist or do not log in a file. But not starting the program because the directory does not exist does not make sense to me.

So for example the start command of cbpi for the dev environment (in launch.json) could look something like this:
`"args": ["--config-folder-path=./.devcontainer/cbpi-dev-config", "--log-folder-path=./.devcontainer/logs" "start"]`

I use docker also for running the environment for the actual brewery. I'd not want the logs to be part of the config because I don't want them to clutter my config which I store in git. Logs are pure runtime info and should not be mixed with the config IMHO. With the config parameter I could simply place the log to another volume or something like that and hopefully never look at it :smile:

Reply:
> I'd not want the logs to be part of the config because I don't want them to clutter my config which I store in git

thats why i want to put it in the parent directory

> If you can not start cbpi if the log directory is not present, maybe that's something that should be changed.

problem with the log folder not being present is that then the sensor data cannot be stored and there is no history available and re is no sensor history stored and no charts etc. So especially for someone starting this on a Pi as intended and as documented and not having charts is a bad thing. So the log folder should be created on `cbpi setup` . And when a config is transferred and the log folder isnt present the `cbpi start` informs you that a `cbpi setup` would resolve the issue and terminates.

I want to implement the `--log-folder-path=` option and when it is not provided i will create a log folder in the parent directory of the config folder. This means in the devcontainer the default config will be in the .devcontainer folder and when the `--config-folder-path=` argument is not provided on `cbpi setup` then the log folder as well as the default config folder will both be created at the current directory.

Reply:
> I want to implement the --log-folder-path= option and when it is not provided i will create a log folder in the parent directory of the config folder. This means in the devcontainer the default config will be in the .devcontainer folder and when the --config-folder-path= argument is not provided on cbpi setup then the log folder as well as the default config folder will both be created at the current directory.

Sounds good. You can easily provide a default value for the option so you don't have to care if it's provided or not. Same is done for the config folder path which defaults to "./config" as it was hardcoded before the option was available. See here:

https://github.com/prash3r/craftbeerpi4/blob/4952861a584224311c4072c78ce15abca8913022/cbpi/cli.py#L228

Thanks for your patience with my comments :slightly_smiling_face: